### ISSUE TITLE
Players no longer able to see one another if bad connection player gets disconnected

### ISSUE THREAD

**Issue by shanzenos** [wtfblub/NetspherePirates#71]
It seems that if someone with bad connection joins and either disconnects or leaves the game but isn't fully disconnected all players lose the ability to see one another in lobbies, and the lobby it initially (the log out or disconnect) happens in always freezes everyone in place. The only solution so far seems to be forcibly restarting the server to disconnect that client completely and it will work again fine.

**Comment by BioAleks** [comment]
:+1:

**Comment by shanzenos** [comment]
Actually, I wonder if this bug has any connection to: https://github.com/wtfblub/NetspherePirates/issues/77

**Comment by BioAleks** [comment]
It's no every time happen.

**Comment by wtfblub** [comment]
#77 #78 pretty sure they're all related to the same bug.

**Comment by shanzenos** [comment]
They are. I tested with a friend all day today. We've tried several things, none of which worked to fix it in the source. I can tell you what we learned about it though that might help towards figuring out the cause of the lockup. For one we added console write to a few locations and found that the relay session is not closing properly: http://puu.sh/ssB7K/c785c94cc1.png Although sometimes when this bug occurs, it doesn't even show that console write message. We also found it's any player that disconnects, regardless of connection strength. And it also ONLY happens if players are disconnecting while playing, if things are calm and everyone is standing still they will usually disconnect fine. But should the players be moving and shooting while disconnecting, it will cause this and #77 Lastly, here's part of a log of the event happening in the server console: http://puu.sh/ssL6Q/a539b647bc.txt
